examples/1d_stencil/1d_stencil_8.cpp

```diff
@@ -575,7 +575,8 @@ stepper_server::space stepper_server::do_work(
     }
 
     // limit depth of dependency tree
-    hpx::sliding_semaphore sem(nd);
+    std::shared_ptr<hpx::sliding_semaphore> sem =
+        std::make_shared<hpx::sliding_semaphore>(nd);
```

Comment:
Wouldn' it be more appropriate to fix the `sliding_semaphore` instead of working around it's deficiencies in user code?

Reply:
Sure! Although, technically, we could demand that the user keeps it alive, I suppose this is exactly the issue you've been fixing on other facilities recently, making sure the relevant data remains alive.

Mind if I take a stab at it? :)